File: 0414.第三大的数/0414-第三大的数.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        i=0
        if len(set(nums))<=2:
            return max(nums)
        else:
            nums=list(set(nums))
            while(len(nums)>0):
                if i>2:
                    break
                max_num=max(nums)
                nums.remove(max_num)
                i+=1
        return max_num

File: 0414.第三大的数/0414-第三大的数.cpp
class Solution {
public:
    int thirdMax(vector<int>& nums) {
        //set默认从小到大排序
        set<int> s;
        //:号可以表示直接遍历整个数组
        //s.erase(it)删除地址为it的元素
        for(int num:nums){
            s.insert(num);
            if(s.size()>3){
                s.erase(s.begin());
            }
        }
        //s.begin()指向第一个数据，s.rbegin()指向最后一个数据
        return s.size()==3? *s.begin():*s.rbegin();
    }
};